tests/scenario_test/test_get_hooks.py

```diff
@@ -12,6 +12,9 @@ def setup_method(self):
         cli_args = [get_hooks.__name__, "--protocol", "message-boundaries", "--boundary", ""]
         self.argv_mock = patch.object(sys, "argv", cli_args)
         self.argv_mock.start()
+        # Prevent unpredictable behavior from import order mismatch
+        if get_hooks.__name__ in sys.modules:
+            del sys.modules[get_hooks.__name__]
```

Comment:
Deleting from `sys.modules` feels a bit less than ideal but for tests I think it's alright. Is this caused by `setup_method` re-importing `get_hooks` for each test case?

I'm slightly curious if importing these modules from within the test runner is an alternative fix, but importing anywhere but the top of a file feels so wrong to me so perhaps it's not worth exploring this!

Reply:
We could try to import this inside the test, I think I was able to make it work. This type of manipulation can cause python to fail if the wrong modules are deleted. Maybe there is a safer way to do this. 🤔 

From what I understand deleting from `sys.modules` can be used as a way to force module reloading.

[sys.module](https://docs.python.org/3/library/sys.html#sys.modules)
> This is a dictionary that maps module names to modules which have already been loaded. This can be manipulated to force reloading of modules and other tricks.

Reply:
Oh dang! I should've checked the official docs haha. If it's noted that tricks can be done with `sys.module`, a lot of my concern disappears!

Totally your call on looking for alternatives to this approach! 🚀

Reply:
Incredible find 🙏 Thanks again for looking into this fix and the follow up!